src/shared/components/date-bubble/index.js

```diff
@@ -1,28 +1,36 @@
 // Displays date bubble
 
-import React, { Component } from 'react';
+import React, { PropTypes } from 'react';
 import styles from './style.css';
 
-export default class DateBubble extends Component {
-  static propTypes = {
-    date: React.PropTypes.string,
-    month: React.PropTypes.string,
-    year: React.PropTypes.string,
-  };
-
-  render() {
-    return (
-      <div className={styles.dateBubble}>
-        <div className={styles.date}>
-          {this.props.date}
-        </div>
-        <div className={styles.month}>
-          {this.props.month}
-        </div>
-        <div className={styles.year}>
-          {this.props.year}
-        </div>
-      </div>
-    );
+const DateBubble = ({
+  startDateTime,
+  endDateTime,
+}) => {
+  let displayDateContent = '';
+  if (endDateTime) {
+    displayDateContent =
+    (`${startDateTime.date} ${startDateTime.monthSym} ${startDateTime.year} - `
+    + `${endDateTime.date} ${endDateTime.monthSym} ${endDateTime.year}`);
+  } else {
+    displayDateContent =
+    `${startDateTime.date} ${startDateTime.monthSym} ${startDateTime.year}`;
   }
-}
+
+  return (<div className={styles.dateBubble}>
+    {displayDateContent}
+  </div>);
+};
+
+const dateShape = {
+  date: PropTypes.string.isRequired,
+  monthSym: PropTypes.string.isRequired,
+  year: PropTypes.string.isRequired,
+};
+
+DateBubble.propTypes = {
+  startDateTime: PropTypes.shape(dateShape).isRequired,
+  endDateTime: PropTypes.shape(dateShape),
```

Comment:
Not anymore - if not provided DateBubble will use this as indication that date range should not be displayed at all